test/paranoia_test.rb

```diff
@@ -758,6 +763,60 @@ def test_missing_restore_recursive_on_polymorphic_has_one_association
     assert_equal 0, polymorphic.class.count
   end
 
+  def test_counter_cache_column_update_on_destroy#_and_restore_and_really_destroy
+    parent_model_with_counter_cache_column = ParentModelWithCounterCacheColumn.create
+    related_model = parent_model_with_counter_cache_column.related_models.create
+
+    assert_equal 1, parent_model_with_counter_cache_column.reload.related_models_count
+    related_model.destroy
+    assert_equal 0, parent_model_with_counter_cache_column.reload.related_models_count
+    # related_model.restore
```

Comment:
I asked about `restore` in the [Notes](https://github.com/radar/paranoia/pull/192#issue-51914917) of the PR's description. Personally I don't need such a behaviour. So removing.